### ISSUE TITLE
Ignore labels for EmbedID

### ISSUE THREAD

**Issue by unnonouno** [pfnet/chainer#832]
Loss functions supports ignore labels, but `EmbedID` does not. When a user set label variable = -1, it causes out-of-bounds problem in EmbedID. It is common problem in language modeling because input variables and output variables for LM are usually shared.

**Comment by cemoody** [comment]
Also true for `NegativeSampling`. The loss functions `sigmoid_cross_entropy` and `softmax_cross_entropy` support ignore labels, but negative sampling does not.

**Comment by unnonouno** [comment]
It's a good suggestion!

**Comment by okuta** [comment]
Supporting ignore labels is completed.
Negative sampling ignore will support #1045
